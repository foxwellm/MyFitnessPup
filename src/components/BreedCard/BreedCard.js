import React, { Component } from 'react'
import { importImages } from '../../helpers/importImages'

export class BreedCard extends Component {

  handleClick = () => {
    if (this.props.location.pathname === '/about-breeds') {
      return
    } else {
      this.props.handleSearchFilter(this.props.breed)
    }
  }

  render() {
    const { active, img, isCold, isRunner, isClimber, breed } = this.props
    const images = importImages();
    const cssClasses = [
      "BreedCard-container",
      active ? "activeCard" : null
    ];
    return (

      <div className={cssClasses.join(' ')} name={breed} onClick={this.handleClick}>
        <div className='breed-card-header'>{breed}</div>
        <div className='breed-img' style={{ backgroundImage: `url(${images[img]})` }}></div>
        <div className='breed-attr'>
          {
            isCold ? <i class="fas fa-snowflake"></i> : <div></div>
          }
          {
            isClimber ? <i class="fas fa-mountain"></i> : <div></div>
          }
          {
            isRunner ? <i class="fas fa-running"></i> : <div></div>
          }
        </div>
      </div>
    )
  }
}

export default BreedCard