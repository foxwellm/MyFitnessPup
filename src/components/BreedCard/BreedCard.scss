.BreedCard-container {
  width: 150px;
  height: 180px;
  padding: 5px;
  border-radius: 5px;
  margin-bottom: 5px;
  background-color: $primary;
  // background-image: url('../../../public/images/siberian-husky.jpg')
  // background-size: 300px 200px;
  // background:  no-repeat center center;
  display: flex;
  flex-direction: column;
  // justify-content: space-between;
  &:hover {
    background-color:$primary-light-hover;
  }
}

.breed-card-header {
  height: 40px;
  margin: 0;
}

.breed-img {
  width: 150px;
  height: 100px;
  background-size: 150px 100px;
}

.breed-attr {
  height: 40px;
}