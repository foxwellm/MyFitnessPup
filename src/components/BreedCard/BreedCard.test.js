import React from 'react'
import { shallow } from 'enzyme'
import BreedCard from './BreedCard'

describe('BreedCard', () => {
  let wrapper
  let mockImg
  let mockLocation
  let mockHandleSearchFilter
  let mockBreed
  let mockActive

  beforeEach(() => {
    mockImg = 'siberian-husky.jpg'
    mockLocation = { pathname: '/about-breeds' }
    mockHandleSearchFilter = jest.fn()
    mockBreed = {
      breed: 'Siberian Husky',
      tag: 'siberian-husky',
      img: 'siberian-husky.jpg',
      isCold: true,
      isRunner: true,
      isClimber: true
    }
    mockActive = true

    wrapper = shallow(
      <BreedCard
        img={mockImg}
        location={mockLocation}
        handleSearchFilter={mockHandleSearchFilter}
        breed={mockBreed}
        active={mockActive}
      />
    )
  });

  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the correct snapshot when the path is not "/about-breeds" ', () => {
    mockLocation = { pathname: '/about' }
    wrapper = shallow(
      <BreedCard
        img={mockImg}
        location={mockLocation}
        handleSearchFilter={mockHandleSearchFilter}
        breed={mockBreed}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleClick', () => {
    it('should not call handleSearchFilter if location.pathname is "/about-breeds" ', () => {
      wrapper.instance().handleClick()
      expect(mockHandleSearchFilter).not.toHaveBeenCalledWith(mockBreed)
    })

    it('should call handleSearchFilter with correct params if ocation.pathname is not "/about-breeds"', () => {
      mockLocation = { pathname: '/about' }
      wrapper = shallow(
        <BreedCard
          img={mockImg}
          location={mockLocation}
          handleSearchFilter={mockHandleSearchFilter}
          breed={mockBreed}
        />
      )
      wrapper.instance().handleClick()
      expect(mockHandleSearchFilter).toHaveBeenCalledWith(mockBreed)
    })
  })
})