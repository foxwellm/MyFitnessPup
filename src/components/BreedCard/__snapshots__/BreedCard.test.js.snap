// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedCard should have match the correct snapshot 1`] = `
<div
  className="BreedCard-container "
  onClick={[Function]}
>
  <div
    className="breed-card-header"
  />
  <div
    className="breed-img"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  >
    <div
      className="breed-attr"
    />
  </div>
</div>
`;
